### PR TITLE
[stable/elasticsearch] non-persistent volume hostPath

### PR DESCRIPTION
**Pull request by pavdmyt**
#### What this PR does / why we need it:

Allows to set a volume of type `hostPath` if data nodes are configured without persistent volumes. By default, if `data.persistence.enabled` set to `false` volume of type `emptyDir` will be created. This is not suitable for scenario when you want to mount a specific host directory for non-persistent volume.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

File: stable/elasticsearch/values.yaml [hunk 1]
```diff
@@ -166,6 +166,10 @@ data:
     name: data
     size: "30Gi"
     # storageClass: "ssd"
+  volumeHostPath:
+    enabled: false
+    path: /data
+    type: ""
```
Comment (onorua) [inline stable/elasticsearch/values.yaml]: why not `emptyDir` as it used to be be default in a previous version?

**Review by onorua — APPROVED** [review]

**Comment by pavdmyt** [comment]
@rendhalver @simonswine @icereval is there any chance to merge this PR?

**Comment by pavdmyt** [comment]
@rendhalver @simonswine @icereval any comments on this PR?

**Comment by pavdmyt** [comment]
@rendhalver @simonswine @icereval any chance to review this PR?

**Pull request closed without merging**